code/espurna/mqtt.ino

```diff
@@ -128,56 +153,143 @@ void _mqttConnect() {
 
         _mqtt.connect();
 
-    #else // not MQTT_USE_ASYNC
+    #else // Using PubSubClient or Arduino-MQTT
 
         bool response = true;
 
-        #if ASYNC_TCP_SSL_ENABLED
+        #if SECURE_CLIENT != SECURE_CLIENT_NONE
 
             bool secure = getSetting("mqttUseSSL", MQTT_SSL_ENABLED).toInt() == 1;
             if (secure) {
                 DEBUG_MSG_P(PSTR("[MQTT] Using SSL\n"));
-                if (_mqtt_client_secure.connect(_mqtt_server.c_str(), _mqtt_port)) {
+                #if SECURE_CLIENT == SECURE_CLIENT_BEARSSL
+                uint16_t requested_mfln = getSetting("mqttScMFLN", MQTT_SECURE_CLIENT_MFLN).toInt();
+                if (requested_mfln) {
+                    bool supported = _mqtt_client_secure.probeMaxFragmentLength(_mqtt_server.c_str(), _mqtt_port, requested_mfln);
+                    DEBUG_MSG_P(PSTR("[MQTT] MFLN buffer size %u supported: %s\n"), requested_mfln, supported ? "YES" : "NO");
+                    if (supported) {
+                        _mqtt_client_secure.setBufferSizes(requested_mfln, requested_mfln);
+                    }
+                }
+                #endif
+
+                int check = getSetting("mqttScCheck", SECURE_CLIENT == SECURE_CLIENT_BEARSSL ? SECURE_CLIENT_CHECK_CA : SECURE_CLIENT_CHECK_FINGERPRINT).toInt();
```

Comment:
The way this is organized prevents BearSSL from using a signature check. It would be preferable to be able to configure this.

Reply:
This is just the default setting: on BearSSL that's the CA check and on AxTLS that's the fingerprint (basically the most secure setting on both implementations). Setting ``mqttScCheck`` to ``SECURE_CLIENT_CHECK_FINGERPRINT`` (1) will allow you to do a fingerprint check.

Reply:
Oh, I see. So presumably, that will have to be brought out to the web UI at some point.

Thanks.

Reply:
Web UI configuration is a good point. What do you think @mcspr, should users be able to upload the certificate which is then written on the SPIFFS? Or just keep a toggle to switch the ``mqttScCheck`` setting (none/FP/CA)?

Reply:
According to discussion in https://github.com/xoseperez/espurna/issues/1839 , it is considered appropriate for all configuration options to be made available via defines.

I understand the idea to be that the defines... Define the DEFAULTS, whereas your implementation enforces a predetermined default.

Reply:
Hmm yes, I agree that it would be better to have that configurable via a define as well. It's just that I copied the behavior from PR #1751 ([1](https://github.com/xoseperez/espurna/pull/1751/files#diff-4668235fd2c44f88c437348e5247399aR101) [2](https://github.com/xoseperez/espurna/pull/1751/files#diff-5727c97832f80d3ac6aedd50dbe75fd6R84
) [3](https://github.com/xoseperez/espurna/pull/1751/files#diff-5727c97832f80d3ac6aedd50dbe75fd6R152
)).

But yeah, that means yet another define (``MQTT_SECURE_VALIDATION`` or something) in general.h. As well as another one for the OTA code (``OTA_SECURE_VALIDATION``). It would be great to have something like JSON/ini config file, as mentioned in https://github.com/xoseperez/espurna/issues/1839#issuecomment-518656529.